### DISCUSSION TITLE
Imagnet pretrained model: training accuracy to zero

### DISCUSSION THREAD

**Discussion by jordanott** [Lightning-AI/pytorch-lightning#13334] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Using the exact code from the [imagenet template](https://github.com/Lightning-AI/lightning/blob/master/examples/pl_domain_templates/imagenet.py), I add the following to the end of the script:

```python 
if __name__ == "__main__":
    import pytorch_lightning

    model = ImageNetLightningModel(
        data_path="path/to/imagenet/data",
        pretrained=True,
        workers=8
    )

    trainer = pytorch_lightning.Trainer(
        gpus=4
    )

    trainer.test(model)

    trainer.fit(model)
```

If I run `trainer.test(model)` I get the performance I expect from a pretrained imagenet model. However, when running `trainer.fit(model)` the training accuracy is zero. 

Are the weights getting reinitialized when training is started? If so, how do I prevent that from happening?

**Comment by rohitgr7** [comment, marked as answer]
> Are the weights getting reinitialized when training is started? If so, how do I prevent that from happening?

they are not, can you check the loss and training logic inside the training step. Or maybe there is an issue on how you are calculating the accuracy.
